supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/HtCondorJobExecutionActor.scala

```diff
@@ -160,43 +190,47 @@ class HtCondorJobExecutionActor(override val jobDescriptor: BackendJobDescriptor
+          case _ => self ! JobExecutionResponse(FailedNonRetryableResponse(jobDescriptor.key,
+            new IllegalStateException("Failed to retrieve job(id) and cluster id"), Option(condorReturnCode)))
         }
 
       case 0 =>
-        log.error(s"Unexpected! Received return code for condor submission as 0, although stderr file is non-empty: {}",
-          File(submitFileStderr).lines)
-        FailedNonRetryableResponse(jobDescriptor.key,
-          new IllegalStateException(s"Execution process failed. HtCondor returned zero status code but non empty stderr file: $condorReturnCode"), Option(condorReturnCode))
+        log.error(s"Unexpected! Received return code for condor submission as 0, although stderr file is non-empty: {}", File(submitFileStderr).lines)
+        self ! JobExecutionResponse(FailedNonRetryableResponse(jobDescriptor.key,
+          new IllegalStateException(s"Execution process failed. HtCondor returned zero status code but non empty stderr file: $condorReturnCode"),
+          Option(condorReturnCode)))
 
       case nonZeroExitCode: Int =>
-        FailedNonRetryableResponse(jobDescriptor.key,
-          new IllegalStateException(s"Execution process failed. HtCondor returned non zero status code: $condorReturnCode"), Option(condorReturnCode))
+        self ! JobExecutionResponse(FailedNonRetryableResponse(jobDescriptor.key,
+          new IllegalStateException(s"Execution process failed. HtCondor returned non zero status code: $condorReturnCode"), Option(condorReturnCode)))
     }
   }
 
-  private def trackTaskToCompletion(jobIdentifier: String): BackendJobExecutionResponse = {
-    val jobReturnCode = Try(extProcess.jobReturnCode(jobIdentifier, returnCodePath)) // Blocks until process completes
+  private def trackTask(jobIdentifier: String): Unit = {
+    val jobReturnCode = Try(extProcess.jobReturnCode(jobIdentifier, returnCodePath))
     log.debug("{} Process complete. RC file now exists with value: {}", tag, jobReturnCode)
 
-    // TODO: Besides return code, do we also need to check based on stderr file?
     jobReturnCode match {
-      case Success(rc) if rc == 0 | runtimeAttributes.continueOnReturnCode.continueFor(rc) => processSuccess(rc)
-      case Success(rc) => FailedNonRetryableResponse(jobDescriptor.key,
-        new IllegalStateException("Job exited with invalid return code: " + rc), Option(rc))
-      case Failure(error) => FailedNonRetryableResponse(jobDescriptor.key, error, None)
+      case Success(None) =>
+        import scala.concurrent.duration._
+        // Job is still running in HtCondor. Check back again after `pollingInterval` seconds
+        context.system.scheduler.scheduleOnce(pollingInterval.seconds, self, TrackTaskStatus(jobIdentifier))
+      case Success(Some(rc)) if rc == 0 | runtimeAttributes.continueOnReturnCode.continueFor(rc) => self ! JobExecutionResponse(processSuccess(rc))
```

Comment:
I couldn't find out how an author can specify that. Could you point me to it?
Either way, that check (for rc == 0 is redundant, since that function (`continueFor`) seems to be now checking for rc == 0 by default  anyway, I'll get rid of that)
(PS: Was looking at the readme we have [here](https://github.com/broadinstitute/cromwell#continueonreturncode))